init.lua

```diff
@@ -0,0 +1,576 @@
+vim.opt.path = { ".", "**" }
+
+-- UI
+if vim.o.textwidth == 0 then
+  vim.opt.colorcolumn = "101"
+else
+  vim.opt.colorcolumn = tostring(vim.o.textwidth + 1)
+end
+
+vim.opt.cursorline = true
+vim.opt.laststatus = 3
+vim.opt.number = true
+vim.opt.relativenumber = true
+vim.opt.scrolloff = 10
+vim.opt.signcolumn = "yes"
+vim.opt.splitbelow = true
+vim.opt.splitright = true
+vim.opt.termguicolors = true
+
+-- Search
+vim.opt.hlsearch = false
+vim.opt.incsearch = true
+vim.opt.ignorecase = true
+vim.opt.smartcase = true
+vim.opt.wildignorecase = true
+
+-- Ripgrep (if available)
+if vim.fn.executable("rg") == 1 then
+  vim.opt.grepformat = "%f:%l:%c:%m"
+  vim.opt.grepprg = "rg --vimgrep -i"
+end
+
+-- Undo history
+vim.opt.undodir = vim.fs.joinpath(vim.fn.stdpath("state"), "undodir")
+vim.opt.undofile = true
+
+-- Fallback colorscheme
+vim.cmd.colorscheme("habamax")
+
+-- Diagnotics
+vim.diagnostic.config({
+  underline = false,
+  severity_sort = true,
+})
+
+---}}}
+
+---{{{ Keymaps
+
+--- This is your leader key.
+---
+--- It will be used for most of your custom keymaps to ensure they don't conflict with anything
+--- that's builtin. It is set to the space key here, but feel free to change it! Many people like
+--- the `,` key as well.
+---
+--- See `:help mapleader` for more information.
+vim.g.mapleader = " "
+
+--- Wrapper function for custom keymaps.
+---
+--- See `:help vim.keymap.set()` for more information.
+local map = function(modes, keys, action, desc)
+  vim.keymap.set(modes, keys, action, { silent = true, desc = desc })
+end
+
+-- File explorer
+map("n", "<Leader>e", vim.cmd.Ex, "Opens netrw.")
+
+-- These two override (in my opinion) bad defaults.
+map("n", "x", "\"_x", "Deletes a single character without yanking it.")
+map("n", "Y", "y$", "Makes `Y` behave more like `D` and `C`.")
+
+-- Yanking and pasting using the system clipboard.
+--
+-- If you're using WSL make sure your Windows clipboard is actually accessible from within WSL.
+-- See `:help clipboard-wsl` for more information.
+--
+-- If you're on Linux make sure to install either `xclip` or `wl-clipboard` (depending on whether
+-- you use X11 or Wayland).
+
+map({ "n", "v" }, "<Leader>y", "\"+y", "Yanks to the system clipboard.")
+map({ "n", "v" }, "<Leader>p", "\"+p", "Pastes from the system clipboard.")
+map({ "n", "v" }, "<Leader>P", "\"+P",
+  "Pastes from the system clipboard without yanking the deleted text.")
+
+-- Stay in visual mode when indenting / dedenting text
+map("x", ">", ">gv", "Indents text in visual mode without losing visual selection.")
+map("x", "<", "<gv", "Dedents text in visual mode without losing visual selection.")
+
+-- Move lines around
+map("n", "<A-j>", ":m .+1<CR>==", "Moves the current line down by 1.")
+map("n", "<A-k>", ":m .-2<CR>==", "Moves the current line up by 1.")
+map("x", "<A-j>", ":m '>+1<CR>=gv", "Moves the current line down by 1.")
+map("x", "<A-k>", ":m '<-2<CR>=gv", "Moves the current line up by 1.")
```

Comment:
Why not map these bad boys to just `H J K L` in visual mode? People might already have something bound to Alt

Reply:
I figured people coming from VSCode for example will be more used to Alt. `J` and `K` also already have functionality in normal mode so the keymap would either have to be different in normal / visual mode or only exist in visual mode.

Reply:
Yeah, kinda true. Let's just leave like that ig